sdk/validator/validator.go

```diff
@@ -89,3 +98,26 @@ func GetRegisteredValidator(
 	}
 	return validatorID, nil
 }
+
+func IsSovereignValidator(
+	network network.Network,
+	subnetID ids.ID,
+	nodeID ids.NodeID,
+) (ValidatorKind, error) {
+	pClient := platformvm.NewClient(network.Endpoint)
+	ctx, cancel := utils.GetAPIContext()
+	defer cancel()
+	vs, err := pClient.GetCurrentValidators(ctx, subnetID, nil)
+	if err != nil {
+		return UndefinedValidatorKind, err
+	}
+	for _, v := range vs {
+		if v.NodeID == nodeID {
+			if v.TxID == ids.Empty {
+				return SovereignValidator, nil
+			}
+			return NonSovereignValidator, nil
+		}
+	}
+	return NonValidator, nil
+}
```

Comment:
we can actually get if sov or not from getSubnet API. If it is sov: permissioned = false

example: https://github.com/ava-labs/avalanche-cli/blob/18af22b0dceefc823bd05432e5e4d05d9b3d7839/cmd/blockchaincmd/add_validator.go#L150